src/MagnumPlugins/KtxImporter/Test/KtxImporterTest.cpp

```diff
@@ -0,0 +1,1334 @@
+
+        CORRADE_VERIFY(!image->isCompressed());
+        CORRADE_COMPARE(image->format(), PixelFormat::RGB8Srgb);
+        CORRADE_COMPARE(image->size(), mipSize);
+
+        const PixelStorage storage = image->storage();
+        /* Alignment is 4 when row length is a multiple of 4 */
+        const Int alignment = ((mipSize[0]*image->pixelSize())%4 == 0) ? 4 : 1;
+        CORRADE_COMPARE(storage.alignment(), alignment);
+        CORRADE_COMPARE(storage.rowLength(), 0);
+        CORRADE_COMPARE(storage.imageHeight(), 0);
+        CORRADE_COMPARE(storage.skip(), Vector3i{});
+
+        CORRADE_COMPARE_AS(image->data(), Containers::arrayCast<const char>(mipViews[i]), TestSuite::Compare::Container);
+
+        mipSize = Math::max(mipSize >> 1, 1);
+    }
+}
+
+void KtxImporterTest::image1DLayers() {
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("KtxImporter");
+    CORRADE_VERIFY(importer->openFile(Utility::Directory::join(KTXIMPORTER_TEST_DIR, "1d-layers.ktx2")));
+
+    const Color3ub data[4*3]{
+        0xff0000_rgb, 0xffffff_rgb, 0x000000_rgb, 0x007f7f_rgb,
+        0xff0000_rgb, 0xffffff_rgb, 0x000000_rgb, 0x007f7f_rgb,
+        0x000000_rgb, 0x000000_rgb, 0x000000_rgb, 0x000000_rgb
+    };
+
+    CORRADE_COMPARE(importer->image2DCount(), 1);
+    CORRADE_COMPARE(importer->image2DLevelCount(0), 1);
+
+    auto image = importer->image2D(0);
+    CORRADE_VERIFY(image);
+
+    CORRADE_VERIFY(!image->isCompressed());
+    CORRADE_COMPARE(image->format(), PixelFormat::RGB8Srgb);
+    CORRADE_COMPARE(image->size(), (Vector2i{4, 3}));
+
+    const PixelStorage storage = image->storage();
+    CORRADE_COMPARE(storage.alignment(), 4);
+    CORRADE_COMPARE(storage.rowLength(), 0);
+    CORRADE_COMPARE(storage.imageHeight(), 0);
+    CORRADE_COMPARE(storage.skip(), Vector3i{});
+
+    CORRADE_COMPARE_AS(image->data(), Containers::arrayCast<const char>(data), TestSuite::Compare::Container);
+}
+
+void KtxImporterTest::image1DCompressed() {
+    auto&& data = CompressedImage1DData[testCaseInstanceId()];
+    setTestCaseDescription(data.name);
+
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("KtxImporter");
+    CORRADE_VERIFY(importer->openFile(Utility::Directory::join(KTXIMPORTER_TEST_DIR, data.file)));
+
+    CORRADE_COMPARE(importer->image1DCount(), 1);
+    CORRADE_COMPARE(importer->image1DLevelCount(0), 1);
+
+    auto image = importer->image1D(0);
+    CORRADE_VERIFY(image);
+
+    CORRADE_VERIFY(image->isCompressed());
+    CORRADE_COMPARE(image->compressedFormat(), data.format);
+    CORRADE_COMPARE(image->size(), data.size);
+
+    const CompressedPixelStorage storage = image->compressedStorage();
+    CORRADE_COMPARE(storage.rowLength(), 0);
+    CORRADE_COMPARE(storage.imageHeight(), 0);
+    CORRADE_COMPARE(storage.skip(), Vector3i{});
+
+    /** @todo Can we test ground truth data here? Probably have to generate it
+              using KtxImageConverter. */
```

Comment:
To answer your other comment as well:

> For compressed formats, how do I best go about getting ground truth data? From a quick glance, `--save-diagnostics` requires `CompareImage`, which doesn't support compressed formats.

`CompareImage` is intended mainly for fuzzy comparison which for compressed formats would mean I need a software decoder for every one of them. It's planned, but pretty far ahead. Another option would be `TestSuite::Compare::StringToFile` which also implements `--save-diagnostic` (yes, a string is shitty, and the console output is also shitty, but it *can* save a file) but that would mean you'd need one ktx file and one raw block data file for each test case, which to me feels a bit excessive.

I'd say do this (since blocks are always 64 or 128 bits anyway, this will always work)

```cpp
    CORRADE_COMPARE_AS(
        Containers::arrayCast<const UnsignedLong>(image.data()),
        Containers::arrayView<UnsignedLong>({}), // probably stored in data.expected instead?
        TestSuite::Compare::Container);
```

run the tests, and grab the ground truth data from the failure diagnostic? :) Assuming you briefly verified the plugin to work externally (opened the image through `magnum-player` for example), this would set things in stone, ensuring that no file reading errors or memory issues happen in the future.

Oh, actually, wouldn't you need this exact sequence of bytes for tests in KtxImageConverter anyway? There you would write it to a file (which you then compare against the ground truth result from PVRTexTool) and what's the input block data there is the output block data in this test.